### PR TITLE
Adding links to searching runs with user_id and life cycle stage

### LINKED ISSUE #6749: [FR] Runs page UI links to initiate search when clicked
### Willingness to contribute

No. I cannot contribute this feature at this time.

### Proposal Summary

Within a runs page, the fields `user_id` and `lifecycle_stage` should be links that, when clicked, execute a search for runs that match the current run's criteria.

For instance, if I am viewing a run initiated by myself, I should be able to click my user_id which will execute a runs search of all other runs that I have initiated in the experiments page view.

### Motivation

> #### What is the use case for this feature?
Improved UX navigation and ease of use.

> #### Why is this use case valuable to support for MLflow users in general?
It prevents users from having to maintain perform as many actions for common tasks for navigation within the UI.

> #### Why is this use case valuable to support for your project(s) or organization?

> #### Why is it currently difficult to achieve this use case?
It is not supported currently.

### Details

Dependent on #6748

### What component(s) does this bug affect?

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/pipelines`: Pipelines, Pipeline APIs, Pipeline configs, Pipeline Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [X] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

### What interface(s) does this bug affect?

- [X] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

### What language(s) does this bug affect?

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

### What integration(s) does this bug affect?

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

### PR DESCRIPTION
**Pull request by jaeday**
Signed-off-by: Jaehyun Shim <jaeday@umich.edu>

## Related Issues/PRs

Resolve #6749

## What changes are proposed in this pull request?
FR for adding links to searching runs based on user_id lifecycle_stage, on RunView.js

## How is this patch tested?

- [ ] Existing unit/integration tests
- [x] New unit/integration tests
- [x] Manual tests (describe details, including test results, below)

https://drive.google.com/file/d/1-lLfzWq-AAA2OVeony91_JhOiPxc7rat/view?usp=sharing - a short demo of the features added.

## Does this PR change the documentation?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly in the documentation preview.

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.

Users are now able to search for runs with user_id and lifecycle_stage upon clicking a hyperlink for user_id and lifecycle_stage in RunView.js

### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [x] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [x] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [x] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Comment by harupy** [comment]
@jaeday Thanks for the PR! Can you take a screen recording and attach that in the issue description?

File: mlflow/server/js/src/experiment-tracking/routes.js [hunk 1]
```diff
@@ -7,6 +7,16 @@ class Routes {
     return `/experiments/${experimentId}`;
   }
 
+  static getExperimentByUser(experimentId, user_id) {
+    const searchString = `searchFilter=user_id:${user_id}`
+    return `/experiments/${experimentId}/${searchString}`;
+  }
+
+  static getExperimentByLifeCycle(experimentId, lifeCycleStage) {
+    const searchString = `searchFilter=lifeCycleStage:${lifeCycleStage}`
```
Comment (harupy) [inline mlflow/server/js/src/experiment-tracking/routes.js:16]:
```suggestion
    const searchString = `lifecycleFilter=${lifeCycleStage}`
```

Comment (jaeday) [reply mlflow/server/js/src/experiment-tracking/routes.js:16]: Hi @harupy , with this change in formatting in line 16, would you like me to format line 11 similarly, like,
const searchString = `user_id=${user_id}`;?

File: mlflow/server/js/src/experiment-tracking/routes.js [hunk 2]
```diff
@@ -7,6 +7,16 @@ class Routes {
...
+    return `/experiments/${experimentId}/${searchString}`;
```
Comment (harupy) [inline mlflow/server/js/src/experiment-tracking/routes.js:17]:
```suggestion
    return `/experiments/${experimentId}?lifecycleFilter=${lifeCycleStage}`;
```

File: mlflow/server/js/src/experiment-tracking/routes.js (hunk 1 above)
Comment (harupy) [reply mlflow/server/js/src/experiment-tracking/routes.js:16]:
```suggestion
```

File: mlflow/server/js/src/experiment-tracking/routes.js [hunk 3]
```diff
@@ -7,6 +7,16 @@ class Routes {
     return `/experiments/${experimentId}`;
   }
 
+  static getExperimentByUser(experimentId, user_id) {
+    const searchString = `searchFilter=user_id:${user_id}`
+    return `/experiments/${experimentId}/${searchString}`;
```
Comment (harupy) [inline mlflow/server/js/src/experiment-tracking/routes.js:12]:
```suggestion
    filter_string = `user_id = ${user_id}`;
    return `/experiments/${experimentId}?searchFilter=${escape(user_id)}`;
```

Comment (jaeday) [reply mlflow/server/js/src/experiment-tracking/routes.js:12]: Wouldn't this be:
```return `/experiments/${experimentId}?searchFilter=${escape(filter_string)}`;``` ?

**Comment by jaeday** [comment]
> @jaeday Thanks for the PR! Can you take a screen recording that demonsrates the feature and attach that in the issue description?

A screen recording is shared on the description of this pr!

File: mlflow/server/js/src/experiment-tracking/routes.js (hunk 3 above)
Comment (harupy) [reply mlflow/server/js/src/experiment-tracking/routes.js:12]: You're right :)

File: mlflow/server/js/src/experiment-tracking/components/RunView.js [hunk 4]
```diff
@@ -340,7 +340,14 @@ export class RunViewImpl extends Component {
               description: 'Label for displaying the user who created the experiment run',
             })}
           >
-            {Utils.getUser(run, tags)}
+            {
```
Comment (harupy) [inline mlflow/server/js/src/experiment-tracking/components/RunView.js:343]:
```suggestion
```

File: mlflow/server/js/src/experiment-tracking/components/RunView.js [hunk 5]
```diff
@@ -371,7 +378,14 @@ export class RunViewImpl extends Component {
                   'Label for displaying lifecycle stage of the experiment run to see if its active or deleted',
               })}
             >
-              {lifecycleStage}
+              {
```
Comment (harupy) [inline mlflow/server/js/src/experiment-tracking/components/RunView.js:381]:
```suggestion
```

File: mlflow/server/js/src/experiment-tracking/components/RunView.js [hunk 6]
```diff
@@ -371,7 +378,14 @@ export class RunViewImpl extends Component {
...
+                <Link
+                  data-test-id='life-cycle-search'
+                  to={Routes.getExperimentByLifeCycle(this.props.experimentId, lifecycleStage)}
+                >
+                  {lifecycleStage}
+                </Link>
+              }
```
Comment (harupy) [inline mlflow/server/js/src/experiment-tracking/components/RunView.js:388]:
```suggestion
```

**Comment by harupy** [comment]
@mlflow-automation autoformat

**Comment by dbczumar** [comment]
@harupy Any updates here?

**Comment by harupy** [comment]
@dbczumar I'll push a commit to update the test, and then merge the PR.

**Review by harupy — APPROVED** [review]
LGTM!

File: mlflow/server/js/src/experiment-tracking/routes.js [hunk 7]
```diff
@@ -7,6 +7,15 @@ class Routes {
     return `/experiments/${experimentId}`;
   }
 
+  static searchRunsByUser(experimentId, user_id) {
+    const filter_string = `user_id = '${user_id}'`;
+    return `/experiments/${experimentId}?searchFilter=${encodeURIComponent(filter_string)}`;
```
Comment (harupy) [inline mlflow/server/js/src/experiment-tracking/routes.js:12]:
```suggestion
    const filterString = `user_id = '${user_id}'`;
    return `/experiments/${experimentId}?searchFilter=${encodeURIComponent(filterString)}`;
```

**Comment by harupy** [comment]
Final check:

https://user-images.githubusercontent.com/17039389/208034219-abad6232-6b19-4bfc-9b8b-a2ec2280b867.mov

**Pull request merged**